### PR TITLE
multicast: add multicast_iface for IPv6 multicast

### PR DESCRIPTION
**Pull request by maximilianfridrich**
Requires:
- https://github.com/baresip/re/pull/1149

### CODE REVIEW

**Comment by maximilianfridrich** [comment]
@sreimers Do you have an idea how this issue could be resolved? It looks like on MacOS there is now a `LIST_FOREACH` macro defined in ` /Applications/Xcode_15.0.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/usr/include/sys/queue.h:499:9`. This will cause issues in BareSIP and re too.

**Comment by sreimers** [comment]
This is an old macro, we simply undef here for example before loading `re_list.h/re.h`

https://github.com/baresip/re/blob/f3b5979d6ad78585b7f6bfd64a78449373ea6330/src/main/main.c#L31-L37

You can try to load `#include <net/if.h>` before re.h include.

**Comment by maximilianfridrich** [comment]
Thank you very much for the quick response! That worked!

File: modules/multicast/multicast.c [hunk 1]
```diff
@@ -309,6 +315,23 @@ static int cmd_mcreg(struct re_printf *pf, void *arg)
 		goto out;
 	}
 
+#ifdef HAVE_GETIFADDRS
+	if (str_isset(mccfg.iface)) {
+		unsigned int if_index;
+		if_index = if_nametoindex(mccfg.iface);
+		if (!if_index) {
+			warning("Can not find interface %s.\n", &mccfg.iface);
```
Comment (cspiel1) [inline modules/multicast/multicast.c:323]: "multicast: could not find interface %s.\n"

Comment (maximilianfridrich) [reply modules/multicast/multicast.c]: Done.

**Pull request merged**